sources/zafira-web/src/main/java/com/qaprosoft/zafira/web/FileUtilController.java

```diff
@@ -66,7 +71,28 @@ public FileUtilController(
     @PostMapping("api/upload")
     @Override
     public String uploadFile(@RequestHeader("FileType") Type type, @RequestParam("file") MultipartFile file) throws IOException {
-        return uploadService.upload(type, file.getInputStream(), file.getOriginalFilename(), file.getSize());
+        String resourceURL;
+
+        // Performing size (less than 2 MB) and file type (JPG/PNG only) validation for images
+        if (COMMON.equals(type) || USERS.equals(type)) {
+
+            boolean isSizeLessThan2Mb = file.getSize() <= 2_000_000;
```

Comment:
Move inside `if()` since it is not used anywhere else, simply check if size is more than allowed to avoid confusion with negation operator (!). Also, i am not sure if 2000000 is 2 MB :)